File: Element/3_Divisors.cxx
// 3 Divisors
// Given a value N. The task is to find how many numbers less than or equal 
// to N have numbers of divisors exactly equal to 3.

#include <bits/stdc++.h>
using namespace std;

bool three_div(int N){
    auto div = 0;
    for(auto x(1); x <= N; x++){
        if(N%x==0)
            div++;
    }
    return(div==3 ? true:false);
}

int main(){
    int N = 30;
    int ct = 0;
    for(auto x(1); x <= N; x++){
        if(three_div(x))
            ct++;
    }
    printf("%d\n",ct); // 3
    return 0;
}

File: Maths/3_Divisors.py
# 3 Divisors
# Given a value N. The task is to find how many numbers less than or equal 
# to N have numbers of divisors exactly equal to 3.

def three_div(N:int)->bool:
	div = 0
	for x in range(1,N+1):
		if(N%x==0):
			div+=1
	f = True if(div==3) else False
	return f

if __name__=='__main__':
	N = 30
	ct = 0
	for x in range(1,N+1):
		if(three_div(x)):
			ct+=1
	print(ct,end="\n") # 3

